flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -632,35 +634,33 @@ private static <P extends ResponseBody> CompletableFuture<P> parseResponse(
         CompletableFuture<P> responseFuture = new CompletableFuture<>();
         final JsonParser jsonParser = objectMapper.treeAsTokens(rawResponse.json);
         try {
-            P response = objectMapper.readValue(jsonParser, responseType);
-            responseFuture.complete(response);
-        } catch (IOException originalException) {
-            // the received response did not matched the expected response type
-
-            // lets see if it is an ErrorResponse instead
-            try {
+            // We make sure it fits to ErrorResponseBody, this condition is enforced by test
```

Comment:
This test is not yet added, because I want to see if the new approach works or not.